src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceState.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+/**
+ * Ip2Geo datasource state
+ *
+ * When data source is created, it starts with PREPARING state. Once the first GeoIP data is generated, the state changes to AVAILABLE.
+ * Only when the first GeoIP data generation failed, the state changes to FAILED.
+ * Subsequent GeoIP data failure won't change data source state from AVAILABLE to FAILED.
```

Comment:
Why wont it change state to failed?

Reply:
The state here is about entire datasource itself not just one failure case during update. When update fails, still a user can use the datasource until it expires. I think I should change the state name from `FAILED` to `CREATE_FAILED`.

Reply:
Let me handle that in the next PR.